accumulator/pollard.go

```diff
@@ -239,6 +234,7 @@ func (p *Pollard) rem2(dels []uint64) error {
 			// fmt.Printf("giving hasher %d %x %x\n",
 			// hn.position, hn.sib.niece[0].data[:4], hn.sib.niece[1].data[:4])
 			hn.dest.data = hn.sib.auntOp()
+			hn.sib.prune()
```

Comment:
this does not quite work since it might prune a node that should become a new root.
is there a way to tell if the hn or it's children are future roots? maybe but swaps are still happening so hard to tell.

Reply:
could also try to traverse the pollard after new roots are selected and prune but it would be cooler to not do that.

Reply:
i think i figured it out. there were hashable nodes above the new roots, but those are not needed. excluding those nodes makes pruning in this place possible.